src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -234,27 +253,93 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 						badRecommendationsString
 					);
 				}
-
 				return validRecommendations;
 			});
 		}
-
 		return TPromise.as([]);
+	}
+
+	private processWorkspaceIgnores(extensionsContent: IExtensionsConfigContent): string[] {
+		const regEx = new RegExp(EXTENSION_IDENTIFIER_PATTERN);
+
+		let countBadIDs = 0;
+		let badIDsString = '';
+		let cleansedIDs = extensionsContent.unwantedRecommendations.filter(element => {
+			if (!regEx.test(element)) {
+				countBadIDs++;
+				badIDsString += `${element} (bad format) Expected: <provider>.<name>\n`;
+				return false;
+			}
+			return true;
+		});
 
+		if (countBadIDs > 0 && this.notificationService) {
+			this.notificationService.warn(
+				'The below ' +
+				countBadIDs +
+				' extension(s) in workspace ignored recommendations have formatting issues:\n' +
+				badIDsString
+			);
+		}
+		return cleansedIDs;
+	}
+
+	refreshAllIgnoredRecommendations(): TPromise<void> {
```

Comment:
Why not update `this._availableRecommendations` here as well? That way, we do need to call `isExtensionAllowedToBeRecommended` from the `__suggest` function